test/crud_test.rb

```diff
@@ -101,6 +106,45 @@ class ModelUpdateOperation < CreateOperation
     ModelUpdateOperation[{id: 1, song: {title: "Mercy Day For Mr. Vengeance"}}].model.must_equal song
   end
 
+  # model Song, :action
+  class FetchCollectionOperation < CreateOperation
+    model Song
+
+    contract do
+      property :title
+    end
+
+    def setup_model!(params)
+      if @user_role == 'admin'
```

Comment:
just to show an example on how differentiate and on fetching collections, you can change to whatever var name you want hehehhe